src/category_theory/monoidal/preadditive.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2021 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.preadditive.additive_functor
+import category_theory.monoidal.category
+
+/-!
+# Preadditive monoidal categories
+
+A monoidal category is `monoidal_preadditive` if it is preadditive and tensor product of morphisms
+is linear in both factors.
+-/
+
+noncomputable theory
+
+namespace category_theory
+
+open category_theory.limits
+open category_theory.monoidal_category
+
+variables (C : Type*) [category C] [preadditive C] [monoidal_category C]
+
+/--
+A category is `monoidal_preadditive` if tensoring is linear in both factors.
+
+Note we don't `extend preadditive C` here, as `abelian C` already extends it,
+and we'll need to have both typeclasses sometimes.
+-/
+class monoidal_preadditive :=
+(tensor_zero : ∀ {W X Y Z : C} (f : W ⟶ X), f ⊗ (0 : Y ⟶ Z) = 0)
+(zero_tensor : ∀ {W X Y Z : C} (f : Y ⟶ Z), (0 : W ⟶ X) ⊗ f = 0)
+(tensor_add : ∀ {W X Y Z : C} (f : W ⟶ X) (g h : Y ⟶ Z), f ⊗ (g + h) = f ⊗ g + f ⊗ h)
+(add_tensor : ∀ {W X Y Z : C} (f g : W ⟶ X) (h : Y ⟶ Z), (f + g) ⊗ h = f ⊗ h + g ⊗ h)
```

Comment:
Should we play the "`obviously` autoparam + `restate_axiom`" game for these fields?

For example, we probably have the correct simp lemmas in the case of tensor products of modules, so tidy should be able to deduce these fields in such cases.